packages/components/src/Button/useButtonStyles.ts

```diff
@@ -0,0 +1,43 @@
+import classnames from "classnames";
+import styles from "./Button.module.css";
+import { ButtonSize, ButtonType, ButtonVariation } from "./Button.types";
+import { useButton } from "./ButtonProvider";
+
+/**
+ * The props that are used to style the button
+ */
+export interface UseButtonStylesProps {
+  size?: ButtonSize;
+  disabled?: boolean;
+  fullWidth?: boolean;
+  loading?: boolean;
+  variation?: ButtonVariation;
+  type?: ButtonType;
+}
+
+export function useButtonStyles({
+  size: sizeProp,
+  disabled,
+  fullWidth,
+  loading,
+  variation = "work",
+  type = "primary",
+}: UseButtonStylesProps) {
+  const { size: contextSize } = useButton();
+  const size = sizeProp || contextSize;
+  const wrapper = classnames(
+    [styles.button, styles[size], styles[variation], styles[type]],
+    {
+      [styles.disabled]: disabled,
+      [styles.fullWidth]: fullWidth,
+      [styles.loading]: loading,
```

Comment:
Might suggest leveraging `:disabled` styling so that if the button is in a fieldset, you get automatic styling.

Reply:
At this time we want to focus on keeping the current behaviour and limit scope by just refactoring to expose our primitives. We want to avoid adding a bunch of small fixes because it can lead to increased scope of testing / verification.
We want to take an iterative approach to this and we can have this be done in follow up tickets